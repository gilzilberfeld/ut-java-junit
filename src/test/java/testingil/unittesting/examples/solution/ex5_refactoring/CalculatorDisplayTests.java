package testingil.unittesting.examples.solution.ex5_refactoring;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

public class CalculatorDisplayTests {

	CalculatorDisplay cd;
	TestLogger log;
	
	@BeforeEach
	public void setup() {
		cd = new CalculatorDisplay();
		log = new TestLogger();
	}

	@Test
	@Disabled()
	public void CharacterizationTests() throws Exception	{
		pressSequence("1");
		pressSequence("12");
		pressSequence("13+");
		pressSequence("11+5");
		pressSequence("21/3");
		pressSequence("15/5=");
		pressSequence("1+2=");
		pressSequence("1C");
		pressSequence("21C");
		pressSequence("C1C");
		Approvals.verify(log.getAll());
	}

	private void pressSequence(String sequence) throws Exception {
		sequence.chars().mapToObj(i -> 
			(char) i).forEach(c -> press (c));
	}

	private void press(char key) {
		cd.press(Character.toString(key));
		log.append(key, cd.getDisplay());
	}

}
