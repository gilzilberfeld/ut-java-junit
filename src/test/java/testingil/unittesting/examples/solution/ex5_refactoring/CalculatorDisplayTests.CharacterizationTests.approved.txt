Pressed 1, Display shows: 1
Pressed 1, Display shows: 11
Pressed 2, Display shows: 112
Pressed 1, Display shows: 1121
Pressed 3, Display shows: 11213
Pressed +, Display shows: 11213
Pressed 1, Display shows: 1
Pressed 1, Display shows: 11
Pressed +, Display shows: 11
Pressed 5, Display shows: 5
Pressed 2, Display shows: 52
Pressed 1, Display shows: 521
Pressed /, Display shows: 521
Pressed 3, Display shows: 3
Pressed 1, Display shows: 31
Pressed 5, Display shows: 315
Pressed /, Display shows: 315
Pressed 5, Display shows: 5
Pressed =, Display shows: 5
Pressed 1, Display shows: 1
Pressed +, Display shows: 1
Pressed 2, Display shows: 2
Pressed =, Display shows: 3
Pressed 1, Display shows: 1
Pressed C, Display shows: 1C
Pressed 2, Display shows: 1C2
Pressed 1, Display shows: 1C21
Pressed C, Display shows: 1C21C
Pressed C, Display shows: E
Pressed 1, Display shows: E
Pressed C, Display shows: E
